mlir/lib/Conversion/RocmlirCustomTosaToLinalg/RocmlirCustomTosaToLinalg.cpp

```diff
@@ -93,6 +97,23 @@ LogicalResult UnsignedCastLoweringPattern::matchAndRewrite(
           assert(inputs.size() == 3);
           result =
               b.create<arith::DivUIOp>(loc, outElemType, inputs[0], inputs[1]);
+        } else if (op.getOperatorName() == "unpack_scale") {
+          assert(inElemType == b.getI32Type());
+          assert(outElemType == b.getF16Type());
```

Comment:
Looks good in general. But it would be nice to have a generic `unpack` operator which would take `input_type` and `elemType` and return array of `elemType[]` instead. 
For example in this case,  `unpack(input)` could return `vector<outElemType>()` and then `scale` would be first element and `bias` would be second element.

Reply:
I think there's a limitation in our codebase regarding linalg::GenericOp number of outputs (for example, look at findPostFusionTransforms in transformMapUtils.cpp). It's assumed the output is always one. We can do a generic unpack(input, element) so that it outputs the output[element].